src/main/resources/templates/pagination.html

```diff
@@ -0,0 +1,20 @@
+<th:block th:fragment="pagination(pageable, page)">
+    <nav aria-label="Page navigation example" th:if="${page?.totalPages > 1}">
+        <ul class="pagination justify-content-end">
+            <li class="page-item" th:classappend="${page.first} ? 'disabled'">
+                <a class="page-link"
+                   th:href="@{/company(page=${page.number - 1}, size=${page.size})}"
```

Comment:
There should not be anything specific to **Company** here. In case needed, pass it through the parameter.